### PR TITLE
scala 2.12

### PR DESCRIPTION
**Pull request by tomrf1**
Also added cloudformation + teamcity + riff-raff support.
I've left the fastly config in s3 for now

### CODE REVIEW

File: cloudformation.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+Resources:
+  Lambda:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub ${App}-${Stage}
+      Description: Fastly cache purger
+      Runtime: java8
+      Handler: com.gu.fastly.Lambda::handle
+      MemorySize: 512
+      Timeout: 15
```
Comment (regiskuckaertz) [inline cloudformation.yaml]: That is very low 😮

File: cloudformation.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      Events:
+        KinesisEvent:
+          Type: EventSourceMapping
+          Properties:
+            EventSourceArn:
+              !Sub arn:aws:kinesis:${AWS::Region}:${AWS::AccountId}:stream/${KinesisStream}
+      CodeUri:
+        Bucket: !Ref DeployBucket
+        Key: !Sub ${Stack}/${Stage}/${App}/${App}.zip
+      Policies:
```
Comment (regiskuckaertz) [inline cloudformation.yaml]: Check out `AWSLambdaExecute` instead, it gives S3 read/write access + logging

File: cloudformation.yaml (hunk 1 above)
Comment (tomrf1) [reply cloudformation.yaml]: indeed, but it's consistent with the existing lambda (which was not cloudformed) so it must be enough!

File: project/plugins.sbt [hunk 3]
```diff
@@ -1,5 +1,6 @@
-addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.3")
+addSbtPlugin("com.typesafe.sbt" % "sbt-native-packager" % "1.3.4")
```
Comment (regiskuckaertz) [inline project/plugins.sbt]: 🎉

File: cloudformation.yaml (hunk 2 above)
Comment (tomrf1) [reply cloudformation.yaml]: that policy is rather liberal with its permissions!

**Pull request merged**